### ISSUE TITLE
The loss seems not consistent between the code and the paper.

### ISSUE THREAD

**Issue by jameszhou-gl** [VLL-HD/GIN#4]
Hi authors,

Thanks for this great work and opensourcing the code.
I have one question: it seems like that the loss function in the code is not consistent with the one in the paper (Eq. (7)).

```python

  if self.empirical_vars:
      # we only need to calculate the std
      sig = torch.stack(
          [z[target == i].std(0, unbiased=False) for i in range(self.n_classes)])
      # negative log-likelihood for gaussian in latent space
      loss = 0.5 + sig[target].log().mean(1) + \
          0.5*np.log(2*np.pi)
  else:
      m = self.mu[target]
      ls = self.log_sig[target]
      # negative log-likelihood for gaussian in latent space
      loss = torch.mean(
          0.5*(z-m)**2 * torch.exp(-2*ls) + ls, 1) + 0.5*np.log(2*np.pi)

```

**Comment by psorrenson** [comment]
Hi, thanks for your interest in the paper and code. 

The second block (when `self.empirical_vars = False`) corresponds to eq. 7 in the paper. This is when the means and log stdevs are learned by the network as parameters.

The first block (when `self.empirical_vars = True`) is another method, where the means and stdevs are not learned, but instead estimated at each iteration from the batch statistics (described in appendix D.1 in the paper). In that appendix it is also described how this is consistent with eq. 7. This was actually the method used in all the experiments. Because the mean and variance are estimated directly from the batch, it is easy to see that the expectation of the first squared term will always be 0.5. This is where the 0.5 in the loss comes from. 

Let me know if you need some further clarification.
